Amp skimlinks e2e tests

slocka opened the pull request:
In order to test amp-skimlinks script, I introduced a new **optional** JSON config to overwrite the request urls used by amp-skimlinks. The config can be inserted inside the `<amp-skimlinks>` tag as follow:

```html
<amp-skimlinks
    layout="nodisplay"
    publisher-code="123X123"
    tracking="true"
>
    <script type="application/json">
    {
        "pageTrackingUrl": "${pageTrackingUrl}",
        "linksTrackingUrl": "${linksTrackingUrl}",
        "nonAffiliateTrackingUrl": "${nonAffiliateTrackingUrl}",
        "waypointUrl": "${waypointUrl}"
    }
    </script>
</amp-skimlinks>
```

This allows us to redirect all the API calls to a proxy which helps us to validate if an API call has happened as expected or not.

Commit:
Draft solution to write amp-skimlinks e2e tests.

For test purposes only, the idea is to pass an optional config by adding a  "<script type="application/json> child element to `<amp-skimlinks>...</amp-skimlinks>.". The config allows us to overwrite the api endpoints in order to pass through a "proxy" while running our e2e tests in the browser.

Commit:
Add e2e tests for amp-skimlinks.

Commit:
Fix linting issues.

Commit:
Add safety test to make sure amp-skimlinks  still works when the config needed only for the tests is not set.

pedroabreu commented on test/integration/test-amp-skimlinks.js:
```diff
@@ -0,0 +1,134 @@
+        expect(data.jv).to.equal(PLATFORM_NAME);
+        expect(data.pub).to.equal('123X123');
+        // nonblocking.io is the default canonical url
+        expect(data.pag).to.equal('http://nonblocking.io/');
```
Would the protocol change depending on the page or is it always `http` ? Could be moved to a constant to avoid repeating it everywhere

slocka replied:
No it's hardcoded to `http`.

pedroabreu commented on test/integration/test-amp-skimlinks.js:
```diff
@@ -0,0 +1,134 @@
+          const match = regex.exec(req.url);
+          expect(match).to.have.lengthOf(2);
+          const data = JSON.parse(decodeURIComponent(match[1]));
+          expect(data.url).to.equal('https://google.com/');
```
Move the url to a constant would avoid repeating it everywhere. Although it's different from the used in the template since it has the trailing slash... 

Happy to keep it this way though, struggle between DRY and easier readability

slocka replied:
I don't think it would bring much to move it to a constant, not worth it.

pedroabreu reviewed (approved):
Two small comments 💅 Looks good 👍

neenu-ram-skim commented:
Looks Good  👍

Commit:
Add e2e tests for affiliation of unknown links.

Commit:
Remove extra line breaks

Commit:
Update tests after removing NA from link tracking

Commit:
Add misssing copyright comment

Commit:
Fix issue with new config change in production build.

Because Google Closure Compiler is renaming properties of objects during the production build, there was a property name mismatch between the config internal object and the config external json object.

Commit:
Move skim options errors to constants

Commit:
Bump amp-skimlinks version number before merging.
